src/app/core/services/label.service.ts

```diff
@@ -109,17 +113,27 @@ export class LabelService {
     this.labelColorMap.clear();
   }
 
-  /**
-   * Converts the (color) hex value into RGB format
-   * @param hex: the hex value
+   /**
+   * Returns true if the given color is considered "dark"
+   * The color is considered "dark" if its luminance is less than COLOR_DARKNESS_THRESHOLD
+   * @param inputColor: the color
    */
-  hexToRgb(hex: string) {
-    const rgbResult = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
-    return rgbResult ? {
-      r: parseInt(rgbResult[1], 16),
-      g: parseInt(rgbResult[2], 16),
-      b: parseInt(rgbResult[3], 16)
-    } : null;
+  isDarkColor(inputColor: string): boolean {
+    const color = (inputColor.charAt(0) === '#') ? inputColor.substring(1, 7) : inputColor;
+    const r = parseInt(color.substring(0, 2), 16);
+    const g = parseInt(color.substring(2, 4), 16);
+    const b = parseInt(color.substring(4, 6), 16);
+    const uiColors = [r / 255, g / 255, b / 255];
+    const c = uiColors.map((col) => {
+      if (col <= 0.03928) {
+        return col / 12.92;
+      }
+      return Math.pow((col + 0.055) / 1.055, 2.4);
+    });
+    // Calculate the luminance of the color
+    const L = (0.2126 * c[0]) + (0.7152 * c[1]) + (0.0722 * c[2]);
+    // A higher threshold value will result in more colors determined to be "dark"
```

Comment:
This comment should be attached to the definition of the constant rather than here?
The comment here should be something like `//the color is "dark" if the luminance is lower than the threshold`